### PR TITLE
[IO][ROOT-9926]Add support for data members which are unique_ptrs<cnst T>

### PR DESCRIPTION
**Pull request by dpiparo**

### CODE REVIEW

File: core/foundation/src/TClassEdit.cxx [hunk 1]
```diff
@@ -1965,6 +1965,10 @@ class NameCleanerForIO {
       // Perhaps we could treat atomics as well like this?
       if (!fMother && TClassEdit::IsUniquePtr(fName+"<")) {
          name = fArgumentNodes.front()->ToString();
+         if (name.find("const ") == 0) {
```
Comment (pcanal) [inline core/foundation/src/TClassEdit.cxx]: What about ("TH1F const" and "TH1F const*")?  [Should we be using something like TSplitType)?

Comment (dpiparo) [reply core/foundation/src/TClassEdit.cxx]: but this is the template of a unique_ptr, does your remark apply?

Comment (pcanal) [reply core/foundation/src/TClassEdit.cxx]: Maybe ... maybe not.   I was think we ought to handle properly also the cases:
```
unique_ptr<Type const>
unique_ptr<Type const*> // yes ought to be rare.
```

**Comment by dpiparo** [comment]
Made sturdier thanks to the suggestion of @pcanal and a unit test for TClassEdit

**Pull request merged**